### ISSUE TITLE
Typescript compiler throws 'Cannot compile external modules unless the '--module' flag is provided.'

### ISSUE THREAD

**Issue by sumitkm** [microsoft/TypeScript#608]
In VS2013 Update 2 I had a class that was exported directly as 

``` javascript
export class ComponentRegistration
{
 public myMethod()
 {
  ...
 }
}
```

This was compiled as an AMD module via the project setting.

However, I downloaded the same code on a machine with VS2013 Update 3 and TypeScript throwing the error

`'Cannot compile external modules unless the '--module' flag is provided.'`

Wrapping the class in a module fixes the error:

``` javascript
module App.Boot
{
 export class ComponentRegistration
 {
  ...
 }
}
```

Given my rudimentary TS knowledge I am assuming the class has incorrect syntax and the TS compiler is a bit confused as to what it should report. 

However, since the AMD module settings are still in place the error is definitely confusing and should be changed to something more appropriate.

**Comment by pligor** [comment]
The flag should be `--module commonjs` or `--module amd` according to this: https://typescript.codeplex.com/wikipage?title=Modules%20in%20TypeScript
